Add description of collections and become_exe keywords

samdoran opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
The descriptions were just missing.

Fixes #68024
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Docs Pull Request
##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
`docs/docsite/keyword_desc.yml`

samdoran committed:
Add description of collections keyword

jborean93 commented on docs/docsite/keyword_desc.yml:
```diff
@@ -7,6 +7,7 @@ always: List of tasks, in a block, that execute no matter if there is an error i
 any_errors_fatal: Force any un-handled task errors on any host to propagate to all hosts and end the play.
 async: Run a task asynchronously if the C(action) supports this; value is maximum runtime in seconds.
 become: Boolean that controls if privilege escalation is used or not on :term:`Task` execution.
+become_exe: Path to the executable used to elevate privileges.
```
Might be a good idea to link to the become plugin page for this or above saying this is implemented by the become plugin itself. I know `runas` has no use for this value.

bcoca commented on docs/docsite/keyword_desc.yml:
```diff
@@ -17,6 +18,11 @@ check_mode: |
+collections: |
+    List of collection namespaces to search for modules.
+
+    .. sealso:: :ref:`using_collections_in_playbook:`
```
```suggestion
    .. seelso:: :ref:`using_collections_in_playbook:`
```

s-hertel commented on docs/docsite/keyword_desc.yml:
```diff
@@ -17,6 +18,11 @@ check_mode: |
+collections: |
+    List of collection namespaces to search for modules.
+
+    .. seelso:: :ref:`using_collections_in_playbook:`
```
```suggestion
    .. seealso:: :ref:`using_collections_in_playbook:`
```

s-hertel commented on docs/docsite/keyword_desc.yml:
```diff
@@ -17,6 +18,11 @@ check_mode: |
     .. seealso:: :ref:`check_mode_dry`
 
+collections: |
+    List of collection namespaces to search for modules.
```
```suggestion
    List of collection namespaces to search for modules and roles.
```
Maybe this should also note that tasks in a role do not inherit the playbook keyword (or that the collections keyword needs to be used in meta/main.yml to apply to tasks within a role?).

samdoran replied:
Good point. I will add a note explaining this.

samdoran committed:
Update based on feedback.

- Add link to become plugins.
- Add note about how the collections keyword works with roles.

acozine commented:
Thanks @samdoran, this looks good! CI ran today and passed, so it's cleared to merge.

acozine commented:
Dammit! Clicked the wrong button.

tpo commented:
Thanks all, this also fixes the [collections link in devel](https://docs.ansible.com/ansible/devel/galaxy/user_guide.html) noted in https://github.com/ansible/ansible/issues/68024#issuecomment-599531037
